src/components/YouTube/useVideos.js

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from 'react';
+
+import mockData from './List/youtube-videos-mock.json';
+
+function useVideos({ search }) {
+  const [videos, setVideos] = useState([]);
+  const [isLoading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    async function fetchVideos() {
+      try {
+        setLoading(true);
+        const url = ['https://www.googleapis.com/youtube/v3/search?'];
+        url.push(`key=${process.env.REACT_APP_YOUTUBE_API_KEY}`);
+        url.push(`&part=snippet&maxResults=10&type=video`);
+        if (search) url.push(`&q=${search}`);
```

Comment:
Took me a while to catch why this was an array, I think that concatenating strings by using the `+` operator is more expressive, at first sight you know that `'asdf' + 'ewer' = 'asdfqwer'` and in this case you have an array of strings, you push items and at the end you do `url.join('')` which I think is somewhat convoluted, at the end of the day it is just preference